File: apps/api/internal/models/todo.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type Todo struct {
	ID             int       `json:"id"`
	UserID         uuid.UUID `json:"user_id"`
	Description    string    `json:"description"`
	AssignedToName string    `json:"assigned_to_name"`
	Completed      bool      `json:"completed"`
	CreatedAt      time.Time `json:"created_at"`
	UpdatedAt      time.Time `json:"updated_at"`
}

File: apps/web/src/types/todo.ts
export interface Todo {
  id: number;
  description: string;
  assigned_to_name: string;
  completed: boolean;
  created_at: string;
  updated_at: string;
}

export interface CreateTodoInput {
  description: string;
  assigned_to_name: string;
}

export type TodoEditable = Pick<Todo, 'id' | 'description' | 'assigned_to_name'>;